src/crypto/crypto_context.cc

```diff
@@ -222,6 +227,272 @@ unsigned long LoadCertsFromFile(  // NOLINT(runtime/int)
   }
 }
 
+enum TrustStatus { UNSPECIFIED, TRUSTED, DISTRUSTED };
+
+std::string stdStringFromCF(CFStringRef s) {
+  if (auto fastCString = CFStringGetCStringPtr(s, kCFStringEncodingUTF8)) {
+    return std::string(fastCString);
+  }
+  auto utf16length = CFStringGetLength(s);
+  auto maxUtf8len = CFStringGetMaximumSizeForEncoding(utf16length,
+                                                      kCFStringEncodingUTF8);
+  std::string converted(maxUtf8len, '\0');
+
+  CFStringGetCString(s, converted.data(), maxUtf8len, kCFStringEncodingUTF8);
+  converted.resize(std::strlen(converted.data()));
+
+  return converted;
+}
+
+std::string getCertIssuer(X509* cert) {
+  ClearErrorOnReturn clearErrorOnReturn;
+  if (cert == nullptr) return {};
+  BIO* bio = BIO_new(BIO_s_mem());
+  if (bio == nullptr) {
+    return nullptr;
+  }
+  if (X509_NAME_print_ex(
+          bio, X509_get_issuer_name(cert), 0, XN_FLAG_ONELINE) <=
+      0) {
+    return {};
+      }
+
+  const int resultLen = BIO_pending(bio);
+  char* issuer = reinterpret_cast<char *>(calloc(resultLen + 1, 1));
+  BIO_read(bio, issuer, resultLen);
+  BIO_free_all(bio);
+
+  std::string str(issuer);
+  return str;
+}
+
+std::string getCertSubject(X509* cert) {
+  ClearErrorOnReturn clearErrorOnReturn;
+  if (cert == nullptr) return {};
+  BIO* bio = BIO_new(BIO_s_mem());
+  if (bio == nullptr) {
+    return nullptr;
+  }
+  if (X509_NAME_print_ex(
+          bio, X509_get_subject_name(cert), 0, XN_FLAG_ONELINE) <=
+      0) {
+    return {};
+      }
+
+  const int resultLen = BIO_pending(bio);
+  char* issuer = reinterpret_cast<char *>(calloc(resultLen + 1, 1));
+  BIO_read(bio, issuer, resultLen);
+  BIO_free_all(bio);
+
+  std::string str(issuer);
```

Comment:
No need to doubly allocate the string, doing it like in `stdStringFromCF` should be fine